### PR TITLE
Drop del-cli dependency

### PR DESCRIPTION
**Pull request by mondeja**
23 packages less to install, just to remove 4 files. Better with a simple script.

### CODE REVIEW

**Comment by LitoMore** [comment]
I'd prefer to keep the `del-cli`. See here https://github.com/sindresorhus/del-cli#comparison

The `del` has safe checks. It's more stable in various scenes. Here is the source code: https://github.com/sindresorhus/del/blob/main/index.js

**Comment by mondeja** [comment]
But we are just removing 4 files that are created by our own script. What safety provides to us in **this** context?

**Comment by mondeja** [comment]
What do you think about this @uncenter?

**Comment by LitoMore** [comment]
> I don't think that the comparison with rimraf is even relevant.

Users may execute the script under another directory.

**Comment by mondeja** [comment]
> > I don't think that the comparison with rimraf is even relevant.
> 
> Users may execute the script under another directory. You're using a relative path for deleting files.

They can always install `del-cli` globally, should be the recommended way. I don't know why we should provide them a CLI to delete files "safely" locally for the repository when we are a icons library, that point doesn't makes sense at all.

**Comment by adamrusted** [comment]
> Users may execute the script under another directory. You're using a relative path for deleting files.

Most of our users, myself included, would just run the script from the VS Code 'NPM Scripts' box at the bottom of their IDE. This would surely run it at the root of the project you're currently in, mitigating the issue we're having. If people are executing the script via the CLI then they should have enough knowledge to parse the error code generated.

**Comment by mondeja** [comment]
Now the paths are absolute, no matter from what directory is executed.

**Comment by LitoMore** [comment]
I still think it did not optimize our project, but it added a `clean.js` that needs to be maintained. I reserve my opinion, but it's up to you to do whatever you want.


BTW, the `npm run clean` fails if there is no file to delete. This needs to be fixed.

**Comment by mondeja** [comment]
> BTW, the npm run clean fails if there is no file to delete. This needs to be fixed.

What do you mean with fix it? The error is handled:

```
$ npm run clean

> simple-icons@11.8.0 clean
> node scripts/build/clean.js

Error cleaning files: ENOENT: no such file or directory, unlink '/.../simple-icons/index.js'
```

File: scripts/build/clean.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+Promise.all([
+  fs.unlink(path.join(rootDirectory, 'index.js')),
+  fs.unlink(path.join(rootDirectory, 'index.mjs')),
+  fs.unlink(path.join(rootDirectory, 'index.d.ts')),
+  fs.unlink(path.join(rootDirectory, 'sdk.js')),
+]).catch((error) => {
+  console.error(`Error cleaning files: ${error.message}`);
+  process.exit(1);
+});
```
Comment (LitoMore) [inline scripts/build/clean.js:21]:
```suggestion
const __dirname = getDirnameFromImportMeta(import.meta.url);
const rootDirectory = path.resolve(__dirname, '..', '..');
const files = ['index.js', 'index.mjs', 'index.d.ts', 'sdk.js'];

Promise.all(
  files.map(file => fs.unlink(path.join(rootDirectory, file)))
).catch((error) => {
  console.error(`Error cleaning files: ${error.message}`);
  process.exit(1);
});
```

I coded on GitHub manually, you may need a reformat with Prettier.

**Comment by LitoMore** [comment]
@mondeja Just let the script always return exit code 0. It doesn’t matter if there are no files to delete.

This script should work fine even if there is only one file. Even there is only one `index.js` file.

**Comment by uncenter** [comment]
Hmm well ideally it would exit 1 on an error outside of the files not existing.

**Comment by mondeja** [comment]
Let me improve it...

**Comment by mondeja** [comment]
Now looks better.

**Comment by LitoMore** [comment]
We should always return exit code 0. Otherwise, it will block subsequent command execution.

As I mentioned: *The script should work fine even if there is only one file. Even there is only one index.js file*.

**Comment by mondeja** [comment]
> We should always return exit code 0. Otherwise, it will block subsequent command execution.
> 
> As I mentioned: _The script should work fine even if there is only one file. Even there is only one index.js file_.

Note that is the current behaviour. You can test it with:

```sh
$ touch index.js
$ node scripts/build/clean.js 
File index.mjs does not exist, skipping...
File index.d.ts does not exist, skipping...
File sdk.js does not exist, skipping...
$ echo $?
0
$ node scripts/build/clean.js 
File index.js does not exist, skipping...
File index.d.ts does not exist, skipping...
File index.mjs does not exist, skipping...
File sdk.js does not exist, skipping...
$ echo $?
0
```

It only returns 1 exit code if there is an error unlinking a file.

**Review by uncenter — APPROVED** [review]

**Comment by LitoMore** [comment]
Great.

**Review by LitoMore — APPROVED** [review]

**Pull request merged**